File: binding/src/charts/libqareaseries.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qareaseries.html
pub const qareaseries = struct {

    /// New constructs a new QAreaSeries object.
    pub fn New() QtC.QAreaSeries {
        return qtc.QAreaSeries_new();
    }


    /// New2 constructs a new QAreaSeries object.
    pub fn New2(upperSeries: ?*anyopaque) QtC.QAreaSeries {
        return qtc.QAreaSeries_new2(@ptrCast(upperSeries));
    }


    /// New3 constructs a new QAreaSeries object.
    pub fn New3(parent: ?*anyopaque) QtC.QAreaSeries {
        return qtc.QAreaSeries_new3(@ptrCast(parent));
    }


    /// New4 constructs a new QAreaSeries object.
    pub fn New4(upperSeries: ?*anyopaque, lowerSeries: ?*anyopaque) QtC.QAreaSeries {
        return qtc.QAreaSeries_new4(@ptrCast(upperSeries), @ptrCast(lowerSeries));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaSeries_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qareaseries.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QAreaSeries_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QAreaSeries_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QAreaSeries_QBaseType(@ptrCast(self));
    }

    pub fn SetUpperSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QAreaSeries_SetUpperSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn UpperSeries(self: ?*anyopaque, ) QtC.QLineSeries {
        return qtc.QAreaSeries_UpperSeries(@ptrCast(self));
    }

    pub fn SetLowerSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QAreaSeries_SetLowerSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn LowerSeries(self: ?*anyopaque, ) QtC.QLineSeries {
        return qtc.QAreaSeries_LowerSeries(@ptrCast(self));
    }

    pub fn SetPen(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QAreaSeries_SetPen(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Pen(self: ?*anyopaque, ) QtC.QPen {
        return qtc.QAreaSeries_Pen(@ptrCast(self));
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QAreaSeries_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QAreaSeries_Brush(@ptrCast(self));
    }

    pub fn SetColor(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QAreaSeries_SetColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn Color(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QAreaSeries_Color(@ptrCast(self));
    }

    pub fn SetBorderColor(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QAreaSeries_SetBorderColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn BorderColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QAreaSeries_BorderColor(@ptrCast(self));
    }

    pub fn SetPointsVisible(self: ?*anyopaque, ) void {
        qtc.QAreaSeries_SetPointsVisible(@ptrCast(self));
    }

    pub fn PointsVisible(self: ?*anyopaque, ) bool {
        return qtc.QAreaSeries_PointsVisible(@ptrCast(self));
    }

    pub fn SetPointLabelsFormat(self: ?*anyopaque, format: []const u8) void {
        const format_str = qtc.libqt_string{
    .len = format.len,
    .data = format.ptr,
};
qtc.QAreaSeries_SetPointLabelsFormat(@ptrCast(self), format_str);
    }

    pub fn PointLabelsFormat(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaSeries_PointLabelsFormat(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qareaseries.PointLabelsFormat: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetPointLabelsVisible(self: ?*anyopaque, ) void {
        qtc.QAreaSeries_SetPointLabelsVisible(@ptrCast(self));
    }

    pub fn PointLabelsVisible(self: ?*anyopaque, ) bool {
        return qtc.QAreaSeries_PointLabelsVisible(@ptrCast(self));
    }

    pub fn SetPointLabelsFont(self: ?*anyopaque, font: ?*anyopaque) void {
        qtc.QAreaSeries_SetPointLabelsFont(@ptrCast(self), @ptrCast(font));
    }

    pub fn PointLabelsFont(self: ?*anyopaque, ) QtC.QFont {
        return qtc.QAreaSeries_PointLabelsFont(@ptrCast(self));
    }

    pub fn SetPointLabelsColor(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QAreaSeries_SetPointLabelsColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn PointLabelsColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QAreaSeries_PointLabelsColor(@ptrCast(self));
    }

    pub fn SetPointLabelsClipping(self: ?*anyopaque, ) void {
        qtc.QAreaSeries_SetPointLabelsClipping(@ptrCast(self));
    }

    pub fn PointLabelsClipping(self: ?*anyopaque, ) bool {
        return qtc.QAreaSeries_PointLabelsClipping(@ptrCast(self));
    }

    pub fn Clicked(self: ?*anyopaque, point: ?*anyopaque) void {
        qtc.QAreaSeries_Clicked(@ptrCast(self), @ptrCast(point));
    }

    pub fn OnClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_Clicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Hovered(self: ?*anyopaque, point: ?*anyopaque, state: bool) void {
        qtc.QAreaSeries_Hovered(@ptrCast(self), @ptrCast(point), state);
    }

    pub fn OnHovered(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, bool) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_Hovered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Pressed(self: ?*anyopaque, point: ?*anyopaque) void {
        qtc.QAreaSeries_Pressed(@ptrCast(self), @ptrCast(point));
    }

    pub fn OnPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_Pressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Released(self: ?*anyopaque, point: ?*anyopaque) void {
        qtc.QAreaSeries_Released(@ptrCast(self), @ptrCast(point));
    }

    pub fn OnReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_Released(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn DoubleClicked(self: ?*anyopaque, point: ?*anyopaque) void {
        qtc.QAreaSeries_DoubleClicked(@ptrCast(self), @ptrCast(point));
    }

    pub fn OnDoubleClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_DoubleClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Selected(self: ?*anyopaque, ) void {
        qtc.QAreaSeries_Selected(@ptrCast(self));
    }

    pub fn OnSelected(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_Selected(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QAreaSeries_ColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_ColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BorderColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QAreaSeries_BorderColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnBorderColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_BorderColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PointLabelsFormatChanged(self: ?*anyopaque, format: []const u8) void {
        const format_str = qtc.libqt_string{
    .len = format.len,
    .data = format.ptr,
};
qtc.QAreaSeries_PointLabelsFormatChanged(@ptrCast(self), format_str);
    }

    pub fn OnPointLabelsFormatChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_PointLabelsFormatChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PointLabelsVisibilityChanged(self: ?*anyopaque, visible: bool) void {
        qtc.QAreaSeries_PointLabelsVisibilityChanged(@ptrCast(self), visible);
    }

    pub fn OnPointLabelsVisibilityChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_PointLabelsVisibilityChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PointLabelsFontChanged(self: ?*anyopaque, font: ?*anyopaque) void {
        qtc.QAreaSeries_PointLabelsFontChanged(@ptrCast(self), @ptrCast(font));
    }

    pub fn OnPointLabelsFontChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_PointLabelsFontChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PointLabelsColorChanged(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QAreaSeries_PointLabelsColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnPointLabelsColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_PointLabelsColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PointLabelsClippingChanged(self: ?*anyopaque, clipping: bool) void {
        qtc.QAreaSeries_PointLabelsClippingChanged(@ptrCast(self), clipping);
    }

    pub fn OnPointLabelsClippingChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QAreaSeries_Connect_PointLabelsClippingChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaSeries_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qareaseries.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaSeries_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qareaseries.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetPointsVisible1(self: ?*anyopaque, visible: bool) void {
        qtc.QAreaSeries_SetPointsVisible1(@ptrCast(self), visible);
    }

    pub fn SetPointLabelsVisible1(self: ?*anyopaque, visible: bool) void {
        qtc.QAreaSeries_SetPointLabelsVisible1(@ptrCast(self), visible);
    }

    pub fn SetPointLabelsClipping1(self: ?*anyopaque, enabled: bool) void {
        qtc.QAreaSeries_SetPointLabelsClipping1(@ptrCast(self), enabled);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAreaSeries_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqareaseries.cpp
#include <QAbstractSeries>
#include <QAreaSeries>
#include <QBrush>
#include <QColor>
#include <QFont>
#include <QLineSeries>
#include <QObject>
#include <QPen>
#include <QPointF>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qareaseries.h>
#include "libqareaseries.h"
#include "libqareaseries.hxx"

QAreaSeries* QAreaSeries_new() {
	 return new VirtualQAreaSeries();
}

QAreaSeries* QAreaSeries_new2(QLineSeries* upperSeries) {
	 return new VirtualQAreaSeries(upperSeries);
}

QAreaSeries* QAreaSeries_new3(QObject* parent) {
	 return new VirtualQAreaSeries(parent);
}

QAreaSeries* QAreaSeries_new4(QLineSeries* upperSeries, QLineSeries* lowerSeries) {
	 return new VirtualQAreaSeries(upperSeries, lowerSeries);
}

libqt_string QAreaSeries_Tr(const char* s) {
	QString _ret = QAreaSeries::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QAreaSeries_Type(const QAreaSeries* self) {
	return self->type();
}

void QAreaSeries_SetUpperSeries(QAreaSeries* self, QLineSeries* series) {
	self->setUpperSeries(series);
}

QLineSeries* QAreaSeries_UpperSeries(const QAreaSeries* self) {
	return self->upperSeries();
}

void QAreaSeries_SetLowerSeries(QAreaSeries* self, QLineSeries* series) {
	self->setLowerSeries(series);
}

QLineSeries* QAreaSeries_LowerSeries(const QAreaSeries* self) {
	return self->lowerSeries();
}

void QAreaSeries_SetPen(QAreaSeries* self, const QPen* pen) {
	self->setPen(*pen);
}

QPen* QAreaSeries_Pen(const QAreaSeries* self) {
	return new QPen(self->pen());
}

void QAreaSeries_SetBrush(QAreaSeries* self, const QBrush* brush) {
	self->setBrush(*brush);
}

QBrush* QAreaSeries_Brush(const QAreaSeries* self) {
	return new QBrush(self->brush());
}

void QAreaSeries_SetColor(QAreaSeries* self, const QColor* color) {
	self->setColor(*color);
}

QColor* QAreaSeries_Color(const QAreaSeries* self) {
	return new QColor(self->color());
}

void QAreaSeries_SetBorderColor(QAreaSeries* self, const QColor* color) {
	self->setBorderColor(*color);
}

QColor* QAreaSeries_BorderColor(const QAreaSeries* self) {
	return new QColor(self->borderColor());
}

void QAreaSeries_SetPointsVisible(QAreaSeries* self) {
	self->setPointsVisible();
}

bool QAreaSeries_PointsVisible(const QAreaSeries* self) {
	return self->pointsVisible();
}

void QAreaSeries_SetPointLabelsFormat(QAreaSeries* self, const libqt_string format) {
	self->setPointLabelsFormat(QString::fromUtf8(format.data, format.len));
}

libqt_string QAreaSeries_PointLabelsFormat(const QAreaSeries* self) {
	QString _ret = self->pointLabelsFormat();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QAreaSeries_SetPointLabelsVisible(QAreaSeries* self) {
	self->setPointLabelsVisible();
}

bool QAreaSeries_PointLabelsVisible(const QAreaSeries* self) {
	return self->pointLabelsVisible();
}

void QAreaSeries_SetPointLabelsFont(QAreaSeries* self, const QFont* font) {
	self->setPointLabelsFont(*font);
}

QFont* QAreaSeries_PointLabelsFont(const QAreaSeries* self) {
	return new QFont(self->pointLabelsFont());
}

void QAreaSeries_SetPointLabelsColor(QAreaSeries* self, const QColor* color) {
	self->setPointLabelsColor(*color);
}

QColor* QAreaSeries_PointLabelsColor(const QAreaSeries* self) {
	return new QColor(self->pointLabelsColor());
}

void QAreaSeries_SetPointLabelsClipping(QAreaSeries* self) {
	self->setPointLabelsClipping();
}

bool QAreaSeries_PointLabelsClipping(const QAreaSeries* self) {
	return self->pointLabelsClipping();
}

void QAreaSeries_Clicked(QAreaSeries* self, const QPointF* point) {
	self->clicked(*point);
}

void QAreaSeries_Connect_Clicked(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QPointF*) = reinterpret_cast<void (*)(QAreaSeries*, const QPointF*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::clicked, [self, slotFunc](const QPointF& point) {
	slotFunc(self, point);
    });
}

void QAreaSeries_Hovered(QAreaSeries* self, const QPointF* point, bool state) {
	self->hovered(*point, state);
}

void QAreaSeries_Connect_Hovered(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QPointF*, bool) = reinterpret_cast<void (*)(QAreaSeries*, const QPointF*, bool)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::hovered, [self, slotFunc](const QPointF& point, bool state) {
	slotFunc(self, point, state);
    });
}

void QAreaSeries_Pressed(QAreaSeries* self, const QPointF* point) {
	self->pressed(*point);
}

void QAreaSeries_Connect_Pressed(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QPointF*) = reinterpret_cast<void (*)(QAreaSeries*, const QPointF*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::pressed, [self, slotFunc](const QPointF& point) {
	slotFunc(self, point);
    });
}

void QAreaSeries_Released(QAreaSeries* self, const QPointF* point) {
	self->released(*point);
}

void QAreaSeries_Connect_Released(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QPointF*) = reinterpret_cast<void (*)(QAreaSeries*, const QPointF*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::released, [self, slotFunc](const QPointF& point) {
	slotFunc(self, point);
    });
}

void QAreaSeries_DoubleClicked(QAreaSeries* self, const QPointF* point) {
	self->doubleClicked(*point);
}

void QAreaSeries_Connect_DoubleClicked(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QPointF*) = reinterpret_cast<void (*)(QAreaSeries*, const QPointF*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::doubleClicked, [self, slotFunc](const QPointF& point) {
	slotFunc(self, point);
    });
}

void QAreaSeries_Selected(QAreaSeries* self) {
	self->selected();
}

void QAreaSeries_Connect_Selected(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*) = reinterpret_cast<void (*)(QAreaSeries*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::selected, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QAreaSeries_ColorChanged(QAreaSeries* self, QColor* color) {
	self->colorChanged(*color);
}

void QAreaSeries_Connect_ColorChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, QColor*) = reinterpret_cast<void (*)(QAreaSeries*, QColor*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::colorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QAreaSeries_BorderColorChanged(QAreaSeries* self, QColor* color) {
	self->borderColorChanged(*color);
}

void QAreaSeries_Connect_BorderColorChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, QColor*) = reinterpret_cast<void (*)(QAreaSeries*, QColor*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::borderColorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QAreaSeries_PointLabelsFormatChanged(QAreaSeries* self, const libqt_string format) {
	self->pointLabelsFormatChanged(QString::fromUtf8(format.data, format.len));
}

void QAreaSeries_Connect_PointLabelsFormatChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const char*) = reinterpret_cast<void (*)(QAreaSeries*, const char*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::pointLabelsFormatChanged, [self, slotFunc](const QString& format) {
	slotFunc(self, format);
    });
}

void QAreaSeries_PointLabelsVisibilityChanged(QAreaSeries* self, bool visible) {
	self->pointLabelsVisibilityChanged(visible);
}

void QAreaSeries_Connect_PointLabelsVisibilityChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, bool) = reinterpret_cast<void (*)(QAreaSeries*, bool)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::pointLabelsVisibilityChanged, [self, slotFunc](bool visible) {
	slotFunc(self, visible);
    });
}

void QAreaSeries_PointLabelsFontChanged(QAreaSeries* self, const QFont* font) {
	self->pointLabelsFontChanged(*font);
}

void QAreaSeries_Connect_PointLabelsFontChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QFont*) = reinterpret_cast<void (*)(QAreaSeries*, const QFont*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::pointLabelsFontChanged, [self, slotFunc](const QFont& font) {
	slotFunc(self, font);
    });
}

void QAreaSeries_PointLabelsColorChanged(QAreaSeries* self, const QColor* color) {
	self->pointLabelsColorChanged(*color);
}

void QAreaSeries_Connect_PointLabelsColorChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, const QColor*) = reinterpret_cast<void (*)(QAreaSeries*, const QColor*)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::pointLabelsColorChanged, [self, slotFunc](const QColor& color) {
	slotFunc(self, color);
    });
}

void QAreaSeries_PointLabelsClippingChanged(QAreaSeries* self, bool clipping) {
	self->pointLabelsClippingChanged(clipping);
}

void QAreaSeries_Connect_PointLabelsClippingChanged(QAreaSeries* self, intptr_t slot) {
    void (*slotFunc)(QAreaSeries*, bool) = reinterpret_cast<void (*)(QAreaSeries*, bool)>(slot);
    QAreaSeries::connect(self, &QAreaSeries::pointLabelsClippingChanged, [self, slotFunc](bool clipping) {
	slotFunc(self, clipping);
    });
}

libqt_string QAreaSeries_Tr2(const char* s, const char* c) {
	QString _ret = QAreaSeries::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QAreaSeries_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAreaSeries::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QAreaSeries_SetPointsVisible1(QAreaSeries* self, bool visible) {
	self->setPointsVisible(visible);
}

void QAreaSeries_SetPointLabelsVisible1(QAreaSeries* self, bool visible) {
	self->setPointLabelsVisible(visible);
}

void QAreaSeries_SetPointLabelsClipping1(QAreaSeries* self, bool enabled) {
	self->setPointLabelsClipping(enabled);
}

// Base class handler implementation
int QAreaSeries_QBaseType(const QAreaSeries* self) {
	auto* vqareaseries = dynamic_cast<const VirtualQAreaSeries*>(self);
	if (vqareaseries && vqareaseries->isVirtualQAreaSeries) {
vqareaseries->setQAreaSeries_Type_IsBase(true);
	return vqareaseries->type();
}
}

// Auxiliary method to allow providing re-implementation
void QAreaSeries_OnType(const QAreaSeries* self, intptr_t slot) {
	auto* vqareaseries = dynamic_cast<const VirtualQAreaSeries*>(self);
	if (vqareaseries && vqareaseries->isVirtualQAreaSeries) {
vqareaseries->setQAreaSeries_Type_Callback(reinterpret_cast<VirtualQAreaSeries::QAreaSeries_Type_Callback>(slot));
}
}

void QAreaSeries_Delete(QAreaSeries* self) {
    delete self;
}

